tests/integration/workflows/go_dep/test_go_dep.py

```diff
@@ -0,0 +1,82 @@
+import os
+import shutil
+import tempfile
+
+from unittest import TestCase
+
+from aws_lambda_builders.builder import LambdaBuilder
+from aws_lambda_builders.exceptions import WorkflowFailedError
+
+
+class TestGoDep(TestCase):
+    TEST_DATA_FOLDER = os.path.join(os.path.dirname(__file__), "data")
+
+    def setUp(self):
+        self.artifacts_dir = tempfile.mkdtemp()
+        self.scratch_dir = tempfile.mkdtemp()
+
+        os.environ["GOPATH"] = self.TEST_DATA_FOLDER
```

Comment:
Don't edit the system's `os.environ`. Use a copy

Or at least unset the value in teardown if you need to edit the system.